### PR TITLE
ROX-24855: Preserve search query params in Compliance Coverage

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR preserves the search query params when switching between the "Checks" and "Clusters" view in Compliance Coverage

## Screen recording

https://github.com/stackrox/stackrox/assets/4805485/f80d34b7-1667-4dcb-b4b2-5b432c44d37b

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfilesTableToggleGroup.tsx [hunk 1]
```diff
@@ -16,11 +17,13 @@ export type ProfilesTableToggleGroupProps = {
 function ProfilesTableToggleGroup({ activeToggle }: ProfilesTableToggleGroupProps) {
     const { navigateWithScanConfigQuery } = useScanConfigRouter();
     const { profileName } = useParams();
+    const location = useLocation();
 
     const handleToggleChange = (resultsView) => {
+        const searchParams = qs.parse(location.search, { ignoreQueryPrefix: true });
```
Comment (bradr5) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfilesTableToggleGroup.tsx:23]: Is there a way to take advantage of `searchFilter` and `getQueryObject`? I haven't tested this but I assume it's going to pass all parameters and I worry that could bite us at some point.

Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfilesTableToggleGroup.tsx:23]: Actually even better, there's a function in `searchUtils` that includes the search prefix `getUrlQueryStringForSearchFilter`

Comment (sachaudh) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfilesTableToggleGroup.tsx:23]: What is the `customParams`?

Comment (sachaudh) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/ProfilesTableToggleGroup.tsx]: I like the idea of using `searchFilter`. That would make it easier. Let me try your suggestions and see how it goes 👍🏼

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx [hunk 2]
```diff
@@ -94,7 +94,11 @@ function ProfileChecksPage() {
     };
 
     function handleProfilesToggleChange(selectedProfile: string) {
-        navigateWithScanConfigQuery(coverageProfileChecksPath, { profileName: selectedProfile });
+        navigateWithScanConfigQuery(
+            coverageProfileChecksPath,
+            { profileName: selectedProfile },
+            searchFilter
```
Comment (bradr5) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileChecksPage.tsx:100]: We actually don't want to preserve filters when swapping profiles, but it's nice to know it's easy to do. Alan brought this up recently and pointed out that preserving a profile check filter in particular wouldn't make sense.

**Review by bradr5 — APPROVED** [review]

**Pull request merged**